src/ol/color.js

```diff
@@ -126,6 +126,43 @@ export function asArray(color) {
   return fromString(color);
 }
 
+/**
+ * @param {Array<number>} hsla HSLA.
+ * @private
+ * @return {Array<number>} RGBA.
+ */
+function hslaToRgba(hsla) {
+  const h = hsla[0] / 360;
+  const s = hsla[1] / 100;
+  const l = hsla[2] / 100;
+  const a = hsla.length > 3 ? hsla[3] : 1;
+
+  function hueToRgb(p, q, t) {
+    t = t < 0 ? t + 1 : t > 1 ? t - 1 : t;
+    return t < 1 / 6
+      ? p + (q - p) * 6 * t
+      : t < 1 / 2
+      ? q
+      : t < 2 / 3
+      ? p + (q - p) * (2 / 3 - t) * 6
+      : p;
+  }
```

Comment:
The `hueToRgb` function could be moved to the module scope instead of being declared each time the `hslaToRgba` function is run.